src/components/Admin/AddQuestion.js

```diff
@@ -0,0 +1,259 @@
+import React, { Component } from "react";
+import styled from "styled-components";
+import { compose } from "recompose";
+
+import Form from "react-bootstrap/Form";
+import Button from "react-bootstrap/Button";
+
+import { withFirebase } from "upe-react-components";
+
+import { isLoggedIn } from "../../util/conditions";
+import Loader from "../Loader";
+import Logo from "../Logo";
+import { Container } from "../../styles/global";
+
+export const RequiredAsterisk = styled.span`
+  color: red;
+  &:after {
+    content: "*";
+  }
+`;
+
+const StyledContainer = styled(Container)`
+  width: 300px;
+  padding: 15px;
+  box-shadow: 0 3px 6px rgba(0, 0, 0, 0.16), 0 3px 6px rgba(0, 0, 0, 0.23);
+  border-radius: 15px;
+  text-align: center;
+  &:hover {
+    -webkit-transform: translateY(-5px);
+    transform: translateY(-5px);
+    transition: all 0.3s linear;
+  }
+`;
+
+const CenteredForm = styled(Form)`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  max-width: 500px;
+  margin: 0 auto;
+`;
+
+class EditQuestion extends Component {
+  _initFirebase = false;
+  state = {
+    fileExtension: "",
+    file: null,
+    name: "",
+    answer: "",
+    description: "",
+    loading: true,
+    validated: false,
+    sending: false,
+    submitted: false,
+    error: null,
+  };
+
+  componentDidMount() {
+    if (this.props.firebase && !this._initFirebase) this.loadData();
+  }
+
+  componentDidUpdate(prevProps) {
```

Comment:
Actually because of this you may need to keep it a class, unless you can figure out a way to do the conditional import inside of a function component.

Reply:
Resolved in commit 328335366bf582d828349cde311f9f15794bbca6